barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.cpp

```diff
@@ -156,20 +152,19 @@ void build_constraints(Builder& builder,
             // The user tells us they how they want these constants set by keeping the nested aggregation object
             // attached to the proof as public inputs. As this is the only object that can prepended to the proof if the
             // proof is above the expected size (with public inputs stripped)
-            std::array<uint32_t, RecursionConstraint::AGGREGATION_OBJECT_SIZE> nested_aggregation_object = {};
+            AggregationObjectIndices nested_aggregation_object = {};
```

Comment:
should actually be pub input indices since the plonk one actually points to the public inputs